es-extension-examples/ext-example-vite/vite.config.js

```diff
@@ -0,0 +1,45 @@
+import reactRefresh from '@vitejs/plugin-react';
+import { defineConfig } from 'vite';
+import dotenv from "dotenv";
+dotenv.config();
+
+const urlPath = process.env.URLPATH || ''
+
+export default defineConfig({
+  /* when in lib mode, must define process.env */
+  define: {
+    'process.env': JSON.stringify({
+      NODE_ENV: process.env.NODE_ENV,
+    })
+  },
+  base: urlPath,
+  root: '.',
+  plugins: [
+    reactRefresh(),
+  ],
+  server: {
+    port: 9098,
+    strictPort: true,
```

Comment:
Using strictPort:true ensure thst Vite will not automatically choose an alternative port when the specified one is unavailable